File: python/p3.py
#What is the largets prime factor of the number 600851475143
import math

def isprime( num ):
	i = 2
	while i < math.sqrt(num):
		if num % i == 0:
			return False
		i+=1
	return True


x = 600851475143
sqrtx = math.floor(math.sqrt(x))

j = sqrtx
while j > 1:
	if x % j == 0 and isprime(j):
		print j
		break
	j-=1

print j

File: c++/p3.cpp
/*
 * File:   p3.cpp
 * Author: Josh
 *
 * Created on May 13, 2012, 4:36 PM
 *
 * The prime factors of 13195 are 5, 7, 13 and 29.
 *
 * What is the largest prime factor of the number 600851475143 ?
 */

#include "p3.hpp"
#include "BigInt.hpp"
#include <iostream>

p3::p3()
{
}

p3::p3(const p3& orig)
{
}

p3::~p3()
{
}

BigInt p3::execute()
{
  BigInt modResult,largestFactor=1,big = "600851475143";
  while( big > 1 )
  {
    for( BigInt i = 2; i <= big; i+=1 )
    {
      modResult = big % i;
      if( modResult == 0 )
      {
        largestFactor = largestFactor > i ? largestFactor : i;
        big /= i;
        break;
      }
    }

  }
  return largestFactor;
}